admin/cli/pkg/cli/request/request.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package request
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+type Client struct {
+	httpClient  *http.Client
+	hostHeader  string
+	bearerToken string
+}
+
+func NewClient(bearerToken string, hostHeader string, insecureSkipVerify bool) *Client {
+	httpClient := &http.Client{}
```

Comment:
check out the debugging round tripper https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/sdp-pipelines?path=/tooling/pkg/clients/ev2/client.go&version=GBmain&line=233&lineEnd=234&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

you likely want something like that for ease of testing

Reply:
i've adopted that pattern